trezorlib/cardano.py

```diff
@@ -0,0 +1,127 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2018 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import base64
+import struct
+import binascii
+from typing import List
+
+from . import messages
+from . import tools
+from .client import CallException
+from .client import field
+from .client import expect
+from .client import session
+
+
+REQUIRED_FIELDS_TRANSACTION = ('inputs', 'outputs', 'transactions')
+REQUIRED_FIELDS_INPUT = ('path', 'prev_hash', 'prev_index', 'type')
+
+
+@expect(messages.CardanoAddress)
+def get_address(client, address_n, show_display=False):
+    return client.call(
+        messages.CardanoGetAddress(
+            address_n=address_n, show_display=show_display))
+
+
+@expect(messages.CardanoPublicKey)
+def get_public_key(client, address_n):
+    return client.call(
+        messages.CardanoGetPublicKey(address_n=address_n)
+    )
+
+
+@expect(messages.CardanoMessageSignature)
+def sign_message(client, address_n, message):
+    return client.call(
+        messages.CardanoSignMessage(
+            address_n=address_n,
+            message=message.encode()
+        )
+    )
+
+
+def verify_message(client, public_key, signature, message):
+    try:
+        response = client.call(
+            messages.CardanoVerifyMessage(
+                public_key=binascii.unhexlify(public_key),
+                signature=binascii.unhexlify(signature),
+                message=message.encode()
+            )
+        )
+
+    except CallException as error:
+        response = error
+
+    if isinstance(response, messages.Success):
+        return True
+
+    return False
+
+
+@session
+def sign_tx(client, inputs: List[messages.CardanoTxInputType], outputs: List[messages.CardanoTxOutputType], transactions: List[bytes]):
+    response = client.call(messages.CardanoSignTransaction(
+        inputs=inputs,
+        outputs=outputs,
+        transactions_count=len(transactions),
+    ))
+
+    while response.tx_index is not None:
+        tx_index = response.tx_index
+
+        transaction_data = binascii.unhexlify(transactions[tx_index])
+        ack_message = messages.CardanoTxAck(transaction=transaction_data)
+        response = client.call(ack_message)
+
+    return (
```

Comment:
```
return response.tx_hash, response.tx_body
```

Reply:
I just returned response because it is CardanoSignedTx message now. Is that ok?